concent_api/core/payments/sci_backend.py

```diff
@@ -0,0 +1,93 @@
+    payment_ts:             int,
+    current_time:           int,
+    to_block:               int,
+    transaction_type:       str,
+):
+    """
+    Function which return list of transactions from payment API
+    where timestamp >= T0
+    """
+    assert isinstance(concent_rpc,              implementation.SCIImplementation)
+    assert isinstance(requestor_eth_address,    str) and len(requestor_eth_address) == 42
+    assert isinstance(provider_eth_address,     str) and len(provider_eth_address)  == 42
+    assert isinstance(payment_ts,               int) and payment_ts     > 0
+    assert isinstance(current_time,             int) and current_time   > 0
+    assert isinstance(to_block,                 int) and to_block       > 0
+    assert isinstance(transaction_type,         str) and transaction_type in ['Force', 'Batch']
+
+    chain_segment_amount    = math.ceil((current_time - payment_ts) / settings.CREATION_NEW_CHAIN_SEGMENT_TIME)
+    from_block              = to_block - chain_segment_amount
+
+    if transaction_type == 'Force':
+        payments_list = concent_rpc.get_forced_payments(
+            requestor_address   = requestor_eth_address,
+            provider_address    = provider_eth_address,
+            from_block          = from_block,
+            to_block            = to_block,
+        )
+    elif transaction_type == 'Batch':
+        payments_list = concent_rpc.get_batch_transfers(
+            payer_address   = requestor_eth_address,
+            payee_address   = provider_eth_address,
+            from_block      = from_block,
+            to_block        = to_block,
+        )
+
+    return payments_list
+
+
+def make_force_payment_to_provider(
+    concent_rpc:            implementation.SCIImplementation,
+    requestor_eth_address:  str,
+    provider_eth_address:   str,
+    value:                  int,
+    payment_ts:             int,
+):
+    """
+    Concent makes transaction from requestor's deposit to provider's account on amount 'value'.
+    If there is less then 'value' on requestor's deposit, Concent transfers as much as possible.
+    """
+    assert isinstance(concent_rpc,              implementation.SCIImplementation)
+    assert isinstance(requestor_eth_address,    str) and len(requestor_eth_address) == 42
+    assert isinstance(provider_eth_address,     str) and len(provider_eth_address)  == 42
+    assert isinstance(payment_ts,               int) and payment_ts   > 0
+    assert isinstance(value,                    int) and value        > 0
+
+    requestor_acc_balance = concent_rpc.get_deposit_value(requestor_eth_address)
+    if requestor_acc_balance < value:
+        value = requestor_acc_balance
+
+    concent_rpc.force_payment(
+        requestor_address   = requestor_eth_address,
+        provider_address    = provider_eth_address,
+        value               = value,
+        closure_time        = payment_ts,
+    )
+
+
+def is_provider_account_status_positive(
```

Comment:
This is called `is_provider_account_status_positive` but uses `requestor_eth_address`. Lets make it independent, just call it `is_account_status_positive` and use `eth_address`.

Reply:
Done